opentelemetry-api/src/opentelemetry/configuration/__init__.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# FIXME find a better way to avoid all those "Expression has type "Any"" errors
+# type: ignore
+
+"""
+Simple configuration manager
+
+This configuration manager reads configuration values from a JSON file, the
+values read there can be overriden by environment variables.
+
+It would probably be better to replace this configuration manager with a more
+powerful one, Dynaconf, for example.
+"""
+
+from json import load
+from os import environ
+from os.path import exists, expanduser, join
+
+
+class Configuration:
+    _instance = None
+
+    __slots__ = ("tracer_provider", "meter_provider")
+
+    def __new__(cls) -> "Configuration":
+        if Configuration._instance is None:
+
+            configuration = {
+                key: "default_{}".format(key) for key in cls.__slots__
+            }
+
+            configuration_file_path = join(
+                expanduser("~"), ".config", "opentelemetry_python.json"
+            )
```

Comment:
We should probably be looking in the project root, not the user's home dir.

Reply:
What's the motivation behind doing this in the $HOME/.config directory? I feel like most applications these days have some deployment mechanism of a package, with it's own configs, that would be inside the directory of the running application.

In my experience anything to do with implicit file expectations gets nasty quickly, and people find themselves having to hack in code to try to adhere to the expectations of the config loader, or hacking the otel code itself to try to change it's behavior.

I don't feel like we have enough data to know what the right solution is around configs defined in markup. But please link to me some motivations if I'm missing context.

Reply:
Hello

Ok, yes, this is pretty much a very rough implementation of this reading a configuration file, just to have a minimal working example. Maybe we can define the path of the configuration file in an environment variable?